### PR TITLE
Add tags support to (regional) instance group manager

### PR DESCRIPTION
**Pull request by pjotrekk**
This PR enables passing input-only field `params.resource_manager_tags` to `resource_compute_instance_group_manager` and `resource_compute_region_instance_group_manager`. It  enables to set Tags to these resources at creation.

Fixes: https://github.com/hashicorp/terraform-provider-google/issues/17616

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added `params.resource_manager_tags` field to `resource_compute_instance_group_manager` and `resource_compute_region_instance_group_manager` that enables to create these resources with tags (beta)
```

### CODE REVIEW

**Comment by askubis** [comment]
LGTM&Approval for resource logic

**Review by shuyama1 — COMMENTED** [review]
Looks like a significant amount of tests are triggers for this change https://github.com/GoogleCloudPlatform/magic-modules/pull/10217#issuecomment-2004662343. I'll test it locally to make sure this change does not affect pre-existing configurations. Just want to be extra cautious since these fields are non-updatable. 

Meanwhile, @pjotrekk would you mind resolving the conflicts in go.mod and go.sum files? Thanks!

**Comment by shuyama1** [comment]
Tested locally and doesn't seem the change affects pre-existing configurations (wonder why multiple tests are triggered in recording though 🤷 ). Mostly LGTM. we'll still need to resolve the conflicts though. Thanks :)

**Comment by pjotrekk** [comment]
> Tested locally and doesn't seem the change affects pre-existing configurations (wonder why multiple tests are triggered in recording though 🤷 ). Mostly LGTM. we'll still need to resolve the conflicts though. Thanks :)

Conflicts resolved. PTAL :)

**Review by askubis — APPROVED** [review]
LGTM!

**Review by shuyama1 — CHANGES_REQUESTED** [review]
Would you mind also rebasing your PR. Sorry we had some issues in our CI earlier. a rebase should pick up the latest fix. Thank you!

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_group_manager_test.go.erb [hunk 1]
```diff
@@ -424,6 +425,32 @@ func TestAccInstanceGroupManager_waitForStatus(t *testing.T) {
 	})
 }
 
+func TestAccInstanceGroupManager_resourceManagerTags(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (pjotrekk) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance_group_manager_test.go.erb:429]: Not a problem, thanks for catching this! Added the version guard

File: mmv1/third_party/terraform/services/compute/resource_compute_region_instance_group_manager_test.go.erb [hunk 2]
```diff
@@ -393,6 +394,32 @@ func TestAccRegionInstanceGroupManager_APISideListRecordering(t *testing.T) {
 	})
 }
 
+func TestAccRegionInstanceGroupManager_resourceManagerTags(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (pjotrekk) [reply mmv1/third_party/terraform/services/compute/resource_compute_region_instance_group_manager_test.go.erb:398]: Done

**Comment by pjotrekk** [comment]
@shuyama1 I rebased and added the guards. PTAL

**Review by shuyama1 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**